agent/engine/dependencygraph/graph.go

```diff
@@ -131,6 +133,46 @@ func verifyStatusResolvable(target *api.Container, existingContainers map[string
 	return true
 }
 
+func verifyTransitionDependenciesResolved(target *api.Container, existingContainers map[string]*api.Container) bool {
+	targetGoal := target.GetDesiredStatus()
+	if targetGoal >= api.ContainerStopped {
+		// A container can always stop, die, or reach whatever other state it
+		// wants regardless of what dependencies it has
+		return true
+	}
+
+	for _, containerDependency := range target.TransitionDependencySet.ContainerDependencies {
+		maybeResolves, exists := existingContainers[containerDependency.Container]
+		if !exists {
+			return false
+		}
+		if !resolvesContainerTransitionDependency(target, maybeResolves, containerDependency) {
+			return false
+		}
+	}
+	return true
+}
+
+func resolvesContainerTransitionDependency(target *api.Container, resource *api.Container, dependency api.ContainerDependency) bool {
+	targetDesired := target.GetDesiredStatus()
+	if targetDesired < dependency.DependentStatus {
+		// not trying to reach dependent status
+		return true
+	}
+	targetKnown := target.GetKnownStatus()
+	if targetKnown >= dependency.DependentStatus {
```

Comment:
When will this happen, if this happened, isn't that a bug? If a container's dependency isn't satisfied, why it can process and has the `targetKnown >= dependency.DependentStatus`? Or you just want to check the knownstatus is >=stopped?

Reply:
If `targetKnown >= dependency.DependentStatus`, that means that the dependency was already satisfied in a previous transition and this just skips checking it again.  It's not a bug.